Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -49,52 +52,291 @@ public FunctionManager(IEnvironment environment, ITraceFactory traceFactory)
+
+                if (template != null)
+                {
+                    await DeployTemplateFromGithub(template, functionDir);
+                }
+                else
+                {
+                    throw new FileNotFoundException($"template: {functionEnvelope.TemplateId} was not found");
+                }
+            }
+            else
+            {
+                // Create or update function
+                if (!FileSystemHelpers.DirectoryExists(functionDir))
+                {
+                    // create a new function
+                    FileSystemHelpers.EnsureDirectory(functionDir);
+                }
+                await FileSystemHelpers.WriteAllTextToFileAsync(Path.Combine(functionDir, Constants.FunctionsConfigFile), JsonConvert.SerializeObject(functionEnvelope?.Config ?? new JObject()));
+            }
+            return await GetFunctionConfig(name);
+        }
+
+        public async Task<IEnumerable<FunctionEnvelope>> ListFunctionsConfig()
+        {
+            var configList = await Task.WhenAll(
+                    FileSystemHelpers
+                    .GetDirectories(_environment.FunctionsPath)
+                    .Select(d => Path.Combine(d, Constants.FunctionsConfigFile))
+                    .Where(FileSystemHelpers.FileExists)
+                    .Select(async f => { try { return await GetFunctionConfig(Path.GetFileName(Path.GetDirectoryName(f))); } catch { return null; } }));
+            return configList.Where(c => c != null);
+        }
+
+        public async Task<FunctionEnvelope> GetFunctionConfig(string name)
+        {
+            var path = Path.Combine(GetFunctionPath(name), Constants.FunctionsConfigFile);
+            if (FileSystemHelpers.FileExists(path))
+            {
+                return CreateFunctionConfig(await FileSystemHelpers.ReadAllTextFromFileAsync(path), name);
+            }
+
+            throw new FileNotFoundException($"Function ({path}) does not exist");
+        }
+
+        public async Task<JObject> GetHostConfig()
+        {
+            var path = Path.Combine(_environment.FunctionsPath, Constants.FunctionsHostConfigFile);
+            if (FileSystemHelpers.FileExists(path))
+            {
+                return JObject.Parse(await FileSystemHelpers.ReadAllTextFromFileAsync(path));
+            }
+
+            throw new FileNotFoundException($"Host file ({path}) does not exist");
+        }
+
+        public async Task<JObject> PutHostConfig(JObject content)
+        {
+            var path = Path.Combine(_environment.FunctionsPath, Constants.FunctionsHostConfigFile);
+            await FileSystemHelpers.WriteAllTextToFileAsync(path, JsonConvert.SerializeObject(content));
+            return await GetHostConfig();
+        }
+
+        public FunctionEnvelope CreateFunctionConfig(string configContent, string functionName)
```

Comment:
It's just the config property that gets written to disk, while the descriptor is always generated dynamically.